### ISSUE TITLE
same number of rows does not seem right

### ISSUE THREAD

**Issue by kontikiman** [jdermody/brightwire#27]
hi, when creating a table as follows 
```
   var builder = BrightWireProvider.CreateDataTableBuilder();
   builder.AddColumn(ColumnType.Matrix, "Past");
   builder.AddColumn(ColumnType.Matrix, "Future");
    //...Build 2 float matrices with same columns and different rows
   builder.Build();//throws
```

The builder throws an exception unless I force the the "future" column to have the same length as the "past value". is this expected?

**Comment by kontikiman** [comment]
How can I implement a graph like this py code more or less:
```

regressor = Sequential()

regressor.add(LSTM(units = 50, return_sequences = True, input_shape = (X_train.shape[1], 1)))
regressor.add(Dropout(0.2))

# Adding a second LSTM layer and some Dropout regularisation
regressor.add(LSTM(units = 50, return_sequences = True))
regressor.add(Dropout(0.2))

# Adding a third LSTM layer and some Dropout regularisation
regressor.add(LSTM(units = 50, return_sequences = True))
regressor.add(Dropout(0.2))
model.add(LSTM(50, activation='relu'))
regressor.BatchNormalisation()

# Adding a fourth LSTM layer and some Dropout regularisation
regressor.add(LSTM(units = 50))
regressor.add(Dropout(0.2))

# Adding the output layer
regressor.add(Dense(units = 1))
# Compiling the RNN
regressor.compile(optimizer = 'adam', loss = 'mean_squared_error')
```

Thank you for bringing a proper ML tool to the .net community

**Comment by jdermody** [comment]
What's the exception? There is some optional validation when building data tables and enforcing tensor conformity is part of that. You can turn it off when creating the data table builder.

However, the code below doesn't throw an exception for me (with validation enabled by default).

```
var builder = BrightWireProvider.CreateDataTableBuilder();
builder.AddColumn(ColumnType.Matrix, "Past");
builder.AddColumn(ColumnType.Matrix, "Future");
builder.Add(FloatMatrix.Create(rowCount:2, columnCount:3), FloatMatrix.Create(rowCount:4, columnCount:3));
builder.Add(FloatMatrix.Create(rowCount:5, columnCount:3), FloatMatrix.Create(rowCount:10, columnCount:3));
var table = builder.Build();
```

**Comment by kontikiman** [comment]
System.ArgumentException: Rows between input and output data tables do not match
   at BrightWire.ExecutionGraph.DataTableAdaptor.SequentialDataTableAdaptor.<>c__DisplayClass1_0.<.ctor>b__0(IRow row, Int32 i) in SequentialDataTableAdaptor.cs:line 29

**Comment by jdermody** [comment]
I think that is actually the correct behavior. When training a classic RNN there should be a corresponding output for each input to derive a gradient at each step.

It sounds like you might be trying to do sequence to sequence instead, in which the lengths can be different. There's a tutorial/explanation here: http://www.jackdermody.net/brightwire/article/Sequence_to_Sequence_with_LSTM
